rxjava2/src/main/java/io/vertx/reactivex/core/json/ObservableUnmarshaller.java

```diff
@@ -0,0 +1,78 @@
+package io.vertx.reactivex.core.json;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.reactivex.ObservableOperator;
+import io.reactivex.Observer;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.disposables.Disposable;
+import io.vertx.core.buffer.Buffer;
+
+import java.io.IOException;
+
+import static io.vertx.core.json.Json.mapper;
+import static java.util.Objects.nonNull;
+
+/**
+ * An operator to unmarshall json to pojos.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class ObservableUnmarshaller<T, B> implements ObservableOperator<T, B> {
```

Comment:
We should avoid using an operator for such simple 1:1 transformation: a `Transformer` is a better fit and would make implementation easier.

This applies to all other RxJava2 types.

In fact, I believe we should also use a `Transformer` in RxJava1 (I'm not sure we do all what's necessary to create a robust operator).